src/Sulu/Bundle/AdminBundle/Resources/js/components/BlockCollection/tests/BlockCollection.test.js

```diff
@@ -406,3 +406,56 @@ test('Should apply renderBlockContent before rendering the block content includi
     expect(blockCollection.find('Block').at(1).prop('children'))
         .toEqual(prefix + value[1].content + typePrefix + 'type1');
 });
+
+test('Should adjust expandedBlocks and generatedBlockIds after updating the value variable', () => {
```

Comment:
Can you also write a test for the other cases (when the same amount resp. more elements are set in `setProps`)?